http.api/src/test/java/ru/tinkoff/qa/neptune/http/api/test/CustomRequestBodyTest.java

```diff
@@ -0,0 +1,214 @@
+
+
+    @BeforeClass
+    public static void beforeClass() {
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(REQUEST_BODY_GSON
+                        )
+                        .withPath(PATH_TO_GSON))
+                .respond(response().withBody(JSON_HAS_BEEN_SUCCESSFULLY_POSTED));
+
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "application/xml")
+                        .withBody(REQUEST_BODY_MAPPED)
+                        .withPath(PATH_TO_JACKSON))
+                .respond(response().withBody(JACKSON_XML_HAS_BEEN_SUCCESSFULLY_POSTED));
+
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "application/xml")
+                        .withBody(REQUEST_BODY_XML_FOR_DOCUMENT)
+                        .withPath(PATH_DOCUMENT_XML))
+                .respond(response().withBody(DOCUMENT_XML_HAS_BEEN_SUCCESSFULLY_POSTED));
+
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "multipart/form-data")
+                        .withBody(REQUEST_BODY_HTML_FOR_DOCUMENT)
+                        .withPath(PATH_DOCUMENT_HTML))
+                .respond(response().withBody(DOCUMENT_HTML_HAS_BEEN_SUCCESSFULLY_POSTED));
+
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "application/x-www-form-urlencoded")
+                        .withBody(REQUEST_BODY_URL_UNLOADED)
+                        .withPath(PATH_URL_UNLOADED))
+                .respond(response().withBody(FORM_HAS_BEEN_SUCCESSFULLY_POSTED));
+
+        clientAndServer.when(
+                request()
+                        .withMethod("POST")
+                        .withHeader("Content-Type", "application/x-www-form-urlencoded")
+                        .withBody(REQUEST_BODY_URL_UNLOADED2)
+                        .withPath(PATH_URL_UNLOADED))
+                .respond(response().withBody(FORM_HAS_BEEN_SUCCESSFULLY_POSTED));
+    }
+
+    @DataProvider
+    public static Object[][] data() throws Exception {
+        var module = new JaxbAnnotationModule();
+
+        var documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        var documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        var inputSource = new InputSource(new StringReader(REQUEST_BODY_XML_FOR_DOCUMENT));
+        var doc = documentBuilder.parse(inputSource);
+
+        var params = new LinkedHashMap<String, String>();
```

Comment:
### If desired:
var params = new LinkedHashMap<String, String>() {{
    put("param1", "value1");
    put("param1", "value1");
}};

Reply:
I think it is not necessary here